zc_plugins/POSM/v6.0.0/admin/includes/classes/observers/class.products_options_stock_admin_observer.php

```diff
@@ -576,6 +580,41 @@ protected function notify_admin_prod_listing_products_query(&$class, string $e,
         $extra_search_fields[] = 'posm.pos_model';
     }
 
+   // -----
+    // Issued by /admin/languages.php; when a new language is added.
+    // Enables POSM to add Back-ordered label for this language in table products_options_stock_names.
+    //
+    protected function notify_admin_language_insert(&$class, string $e, &$insert_id)
+    {
+        global $db;
+
+        // create additional products option stock names records
+        $products_option_stock_names = $db->Execute("SELECT pos_name_id, pos_name
+                                       FROM " . TABLE_PRODUCTS_OPTIONS_STOCK_NAMES . "
+                                       WHERE language_id = " . (int)$_SESSION['languages_id']);
+
+        foreach ($products_option_stock_names as $option_stock_name) {
+          $db->Execute("INSERT INTO " . TABLE_PRODUCTS_OPTIONS_STOCK_NAMES . " (pos_name_id, language_id, pos_name)
+                        VALUES ('" . (int)$option_stock_name['pos_name_id'] . "',
+                                '" . (int)$insert_id . "',
+                                '" . zen_db_input($option_stock_name['pos_name']) . "')");
```

Comment:
Use this instead.  Using INSERT IGNORE in case there's already an entry (otherwise, there will be a duplicate key-set and MySQL Fatal error).  Don't use single-quotes around integer values.
```php
            $db->Execute(
                "INSERT IGNORE INTO " . TABLE_PRODUCTS_OPTIONS_STOCK_NAMES . "
                    (pos_name_id, language_id, pos_name)
                 VALUES
                    (" . $option_stock_name['pos_name_id'] . ", $insert_id, '" . zen_db_input($option_stock_name['pos_name']) . "')"
            );
```